python/lsst/ts/cbp/mock_server.py

```diff
@@ -201,10 +201,15 @@ async def do_new_azimuth(self, azimuth):
         str
         """
         try:
-            self.encoders.AZIMUTH.set_position(float(azimuth))
+            self.encoders.azimuth.set_position(float(azimuth))
             return ""
         except ValueError:
-            raise  # figure out what the real controller does
+            if float(azimuth) > self.encoders.azimuth.max_position:
```

Comment:
What if neither of these conditions is true? I think you are assuming that cannot happen, but that seems unsafe to me, and the fact that your code silently does nothing if that occurs seems very unsafe. You will never know about the problem.

I suggest again that you cap the value before calling set_position. Then no exception should be raised and you need not try to catch it. Here are two of many approaches:
```
def set_constrained_position(self, value, actuator):
   """Set an actuator to a position that is silently constrained to be in bounds.

    Parameters
    ----------
    value : `float`
        Desired value
    actuator : `lsst.ts.simactuators.PointToPointActuator
        The actuator to set.
    """
    constrained_value = min(max(value, actuator.min_position), actuator.max_position)
    self.actuator.set_position(constrained_value)

def do_new_azimuth(self, azimuth):
    """..."""
    self.set_constrained_position(value=float(azimuth), actuator=self.encoders.azimuth)
    return ""
```
Or use a free function to constrain the value, if you prefer. This is arguably a bit cleaner, but it requires a lot of repetition in the calling code:
```
def constrain_value(value, min_value, max_value):
    """..."""
    if max_value <= min_value:
        raise ValueError(f"min_value={min_value} must be < max_value={max_value}")
    return min(max(value, min_value), max_value)

def do_new_azimuth(self, azimuth):
    """..."""
    constrained_azimuth = constrain_value(
        value=float(azimuth),
        min_value= self.encoders.azimuth.min_position,
        max_value= self.encoders.azimuth.max_position,
    )
    self.encoders.azimuth.set_position(constrained_azimuth)
    return ""
````